pkg/controller/jaeger/jaeger_controller.go

```diff
@@ -352,3 +356,75 @@ func (r ReconcileJaeger) getSecretsForNamespace(secrets []corev1.Secret, namespa
 	}
 	return secretsForNamespace
 }
+
+// SyncOnJaegerChanges sync deployments with sidecars when a jaeger CR changes
+func (r *ReconcileJaeger) SyncOnJaegerChanges(object client.Object) []reconcile.Request {
```

Comment:
As far as I understand this function is watching Jaeger CRs for changes, and triggers deployment/namespace reconciliations using a modified annotation.
Why do we need this,  Could be a part of the Jaeger CRs  reconciliation? what is the difference?

Reply:
Sure we can move it up. the function exists only because I have tried to change as little as possible in the existing behavior. But it makes sense. especially considering the [potential error](https://github.com/jaegertracing/jaeger-operator/pull/1828#discussion_r837041904) when the deployment state has changed.